mockwebserver-junit5/src/test/java/mockwebserver3/junit5/ExampleMockWebServerTest.kt

```diff
@@ -0,0 +1,16 @@
+package mockwebserver3.junit5
+
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.RegisterExtension
+import kotlin.test.assertTrue
+
+class ExampleMockWebServerTest {
+  @RegisterExtension
```

Comment:
I was thinking we might be able to implement this in a way where the field you declare is a normal MockWebServer instance. (Though it might be more difficult to implement.)

Reply:
Anything here is possible, it's probably more relevant, what is more idiomatic junit5 test code.  We can try out a few options, see what the tests are like.  Will try your suggestion on this branch tonight.

Reply:
Seems like that is the approach using ParameterResolver here

https://github.com/Dev-Mind/mockwebserver/blob/master/src/main/java/fr/devmind/mockwebserver/MockWebServerExtension.java

cc @javamind

Reply:
@anuraaga any feelings one way or another on this?

Reply:
I see both `@ExtendsWith` with type resolution and `RegisterExtension` fields like this one. The latter has some flexibility by making it easier to configure the test-specific behavior if there's any settings I believe, something like 

```java
private MockWebServerExtension server = new MockWebServerExtension() {
  @Override
  int responseDelay() {
    return 100;
  }
}
```

It's also nice if the extension can be applied beforeAll or beforeEach at user discretion

Reply:
The ExampleMockWebServerParameterTest test shows ExtendsWith. Agree about extending or configuring the instance, as much as we choose to make that public API.

But not sure how to achieve what you are suggesting for beforeAll and beforeEach.  some flag?  Seems a bit clunky and at that point maybe you want your own Extension with your particular lifecycle.

Reply:
I use this pattern to let one option on the extension control it - it's exposed as a protected method for anonymous subclass overriding, which is admittedly not the most beautiful but seems reasonable at the same time.

https://github.com/line/armeria/blob/master/junit5/src/main/java/com/linecorp/armeria/testing/junit5/common/AbstractAllOrEachExtension.java

With many many tests, the ability to reuse the port across tests in a class does become significant in my experience so whether default or not I do hope it's supported easily enough :)

Reply:
This option doesn't feel in the style of OkHttp.  Also as far as public APIs go, I think we already have a wide one in terms of MockWebServer, adding more at an intermediate layer feels like overkill here.  You can already configure teh mockwebserver before it starts.